File: src/main/java/org/funivan/lologa/algo/ga/genome/value/MaxScoreValue.java
package org.funivan.lologa.algo.ga.genome.value;

import org.funivan.lologa.tile.Score.MaxScore;
import org.funivan.lologa.tiles.TilesInterface;

public class MaxScoreValue implements ValueInterface {
    @Override
    public String type() {
        return "max_score";
    }

    @Override
    public Double value(TilesInterface original, TilesInterface tiles) {
        return (double) 1 - ((double) new MaxScore(original).value() / (double) new MaxScore(tiles).value());
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/value/MaxScoreValue.kt
package org.funivan.lologa.algo.ga.genome.value

import org.funivan.lologa.tile.Score.MaxScore
import org.funivan.lologa.tiles.TilesInterface

class MaxScoreValue : ValueInterface {
    override fun type(): String {
        return "max_score"
    }

    override fun value(original: TilesInterface, tiles: TilesInterface): Double {
        return 1.toDouble() - MaxScore(original).value().toDouble() / MaxScore(tiles).value().toDouble()
    }
}
